script/model/model_server.py

```diff
@@ -0,0 +1,249 @@
+#!/usr/bin/env python3
+"""
+This file contains the python model server implementation.
+
+The server should be stateless but with caching of models.
+The message format that the model-server expects should be kept consistent with Messenger class in
+the noisepage source code.
+The command format should be kept with ModelServerManager in the noisepage souce
+code.
+
+TODO(Ricky):
+- Encapsulate the modelserver in a class
+- Modelserver side of callbacks (once we are more clear on the actions)
+- Model invocation could be improved.
+
+"""
+
+from __future__ import annotations
+import zmq
+import sys
+import atexit
+import time
+from enum import Enum, auto
+from typing import Dict, Optional, Tuple
+import json
+import pprint
+import pickle
+import numpy as np
+from data_class import opunit_data
+from sklearn import model_selection
+from mini_trainer import MiniTrainer
+from util import logging_util
+import logging
+
+
+logging_util.init_logging('info')
+# Initialize ZMQ connection with the c++ side
+end_point = sys.argv[1]
+context = zmq.Context()
+socket = context.socket(zmq.DEALER)
```

Comment:
Seems to work if you replace with the following:

```
socket.set_string(zmq.IDENTITY, 'model')
logging.info(f"Python model trying to connect to manager at {end_point}")
socket.connect(f"ipc://{end_point}")
logging.info(f"Python model connected to {end_point}")
socket.send_multipart([''.encode('utf-8'), '0-0-mario'.encode('utf-8')])
```

Inside your MSM, you are asking it to send a message to "model". If you do not set the identity of the Python MS to "model", then it says host unreachable because there is nobody called "model".